File: apps/edifice-vision/src/types/announcement.ts
/** 公告优先级：0-普通 / 1-重要 / 2-紧急 */
export const ANNOUNCEMENT_PRIORITY_MAP: Record<number, string> = {
  0: "普通",
  1: "重要",
  2: "紧急",
};

/** 公告状态：0-草稿 / 1-已发布 / 2-已下线 */
export const ANNOUNCEMENT_STATUS_MAP: Record<number, string> = {
  0: "草稿",
  1: "已发布",
  2: "已下线",
};

export interface AnnouncementVo {
  announcementId: string;
  title: string;
  content: string;
  /** 0-普通 / 1-重要 / 2-紧急 */
  priority: number;
  /** 0-草稿 / 1-已发布 / 2-已下线 */
  status: number;
  publishTime: string | null;
  expireTime: string | null;
  publishUserId: string | null;
  publishUserName: string | null;
  createdTime: string;
  updatedTime: string;
}

export interface CreateAnnouncementParams {
  title: string;
  content: string;
  priority?: number;
  /** 0-草稿 / 1-立即发布 */
  status?: number;
  expireTime?: string;
}

export interface UpdateAnnouncementParams {
  announcementId: string;
  title?: string;
  content?: string;
  priority?: number;
  expireTime?: string;
}

export interface GetAnnouncementListParams {
  keywords?: string;
  status?: number;
  priority?: number;
  current?: number;
  pageSize?: number;
}

File: apps/edifice-core/src/main/java/com/qsy/edifice/domain/entity/Announcement.java
package com.qsy.edifice.domain.entity;

import com.baomidou.mybatisplus.annotation.*;
import com.fasterxml.jackson.databind.annotation.JsonSerialize;
import com.fasterxml.jackson.databind.ser.std.ToStringSerializer;
import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;
import lombok.NoArgsConstructor;

import java.io.Serial;
import java.io.Serializable;
import java.time.LocalDateTime;

/**
 * 公告实体类
 */
@Data
@Builder
@NoArgsConstructor
@AllArgsConstructor
@TableName("announcement")
public class Announcement implements Serializable {

    @Serial
    private static final long serialVersionUID = 1L;

    /**
     * 公告id
     */
    @TableId(value = "announcement_id", type = IdType.ASSIGN_ID)
    @JsonSerialize(using = ToStringSerializer.class)
    private Long announcementId;

    /**
     * 公告标题
     */
    @TableField("title")
    private String title;

    /**
     * 公告内容
     */
    @TableField("content")
    private String content;

    /**
     * 优先级：0-普通/1-重要/2-紧急
     */
    @TableField("priority")
    private Integer priority;

    /**
     * 状态：0-草稿/1-已发布/2-已下线
     */
    @TableField("status")
    private Integer status;

    /**
     * 发布时间
     */
    @TableField("publish_time")
    private LocalDateTime publishTime;

    /**
     * 过期时间
     */
    @TableField("expire_time")
    private LocalDateTime expireTime;

    /**
     * 发布人id
     */
    @TableField("publish_user_id")
    @JsonSerialize(using = ToStringSerializer.class)
    private Long publishUserId;

    /**
     * 创建时间
     */
    @TableField(value = "created_time", fill = FieldFill.INSERT)
    private LocalDateTime createdTime;

    /**
     * 更新时间
     */
    @TableField(value = "updated_time", fill = FieldFill.INSERT_UPDATE)
    private LocalDateTime updatedTime;

    /**
     * 逻辑删除
     */
    @TableField("is_delete")
    @TableLogic
    private Integer isDelete;
}
